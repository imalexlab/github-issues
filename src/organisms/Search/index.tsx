import React, { useCallback, useEffect, useState } from 'react';
import styled from 'styled-components';

const Wrapper = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #565c88;
  margin: 0 auto;
  width: 75%;

  @media (max-width: 768px) {
    width: 100%;
  }
`;

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

const Label = styled.label`
  font-size: 2rem;
  margin-top: 2rem;
  text-align: center;
`;

const Input = styled.input`
  margin: 1rem;
  padding: 10px;
  font-size: 1rem;
  border: 1px solid #565c88;
  background: transparent;
  border-radius: 5px;
`;

const ButtonWrapper = styled.div`
  display: flex;
`;

const Button = styled.button`
  margin: 1rem;
  padding: 10px;
  font-size: 1rem;
  border: 1px solid #565c88;
  background: transparent;
  color: #565c88;
  border-radius: 5px;

  :hover {
    border: 1px solid #565c88;
    background: #565c88;
    color: #f4f5fa;
  }

  :disabled {
    color: #bdc5ff;
    border: 1px solid #bdc5ff;
    background: transparent;
    cursor: not-allowed;
  }
`;

interface Props {
  handleSearch: ({ owner, repo }: { owner: string; repo: string }) => void;
  handleReset: () => void;
  owner: string;
  repo: string;
}

export default function Search({
  handleSearch,
  handleReset,
  owner,
  repo,
}: Props) {
  const [currentOwner, setCurrentOwner] = useState('');
  const [currentRepo, setCurrentRepo] = useState('');

  useEffect(() => {
    setCurrentOwner(owner);
    setCurrentRepo(repo);
  }, [owner, repo]);

  const onSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    handleSearch({ owner: currentOwner, repo: currentRepo });
  };

  const onReset = (event: React.FormEvent) => {
    event.preventDefault();
    setCurrentOwner('');
    setCurrentRepo('');
    handleReset();
  };

  const changeOwner = useCallback((e) => {
    setCurrentOwner(e.target.value);
  }, []);

  const changeRepo = useCallback((e) => {
    setCurrentRepo(e.target.value);
  }, []);

  return (
    <>
      <Wrapper onSubmit={onSubmit} onReset={onReset}>
        <Label htmlFor="issues">
          Search with the name of the repository's owner and the repository name
        </Label>
        <Container>
          <Input
            data-testid="input-owner"
            type="text"
            id="owner"
            aria-label="owner"
            onChange={changeOwner}
            value={currentOwner}
            placeholder="facebook"
          />
          <Input
            data-testid="input-repo"
            type="text"
            id="repository"
            aria-label="repository"
            onChange={changeRepo}
            value={currentRepo}
            placeholder="react"
          />
          <ButtonWrapper>
            <Button
              type="submit"
              data-testid="search"
              onClick={onSubmit}
              disabled={currentOwner === '' || currentRepo === ''}
            >
              Search
            </Button>
            <Button type="reset" onClick={onReset}>
              Reset
            </Button>
          </ButtonWrapper>
        </Container>
      </Wrapper>
    </>
  );
}
