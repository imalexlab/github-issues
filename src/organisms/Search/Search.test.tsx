import React from 'react';
import { render, fireEvent, getByTestId } from '@testing-library/react';
import Search from '.';

test('renders Search', () => {
  const props = {
    handleSearch: jest.fn(),
    handleReset: jest.fn(),
    owner: '',
    repo: '',
  };
  const { container } = render(<Search {...props} />);
  const inputOwner = getByTestId(container, 'input-owner');
  const inputRepo = getByTestId(container, 'input-repo');
  const searchButton = getByTestId(container, 'search');

  expect(searchButton).toBeDisabled();

  
  fireEvent.change(inputOwner, { target: { value: 'facebook' } });
  fireEvent.change(inputRepo, { target: { value: 'react' } });

  expect(inputOwner).toHaveValue('facebook');
  expect(inputRepo).toHaveValue('react');
  expect(searchButton).not.toBeDisabled();
  expect(container).toMatchSnapshot();
});
