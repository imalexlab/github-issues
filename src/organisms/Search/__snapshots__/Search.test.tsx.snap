// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search 1`] = `
<div>
  <form
    class="sc-AxjAm kELFtL"
  >
    <label
      class="sc-AxiKw dXUvmq"
      for="issues"
    >
      Search with the name of the repository's owner and the repository name
    </label>
    <div
      class="sc-AxirZ eFbayh"
    >
      <input
        aria-label="owner"
        class="sc-AxhCb enTIxr"
        data-testid="input-owner"
        id="owner"
        placeholder="facebook"
        type="text"
        value="facebook"
      />
      <input
        aria-label="repository"
        class="sc-AxhCb enTIxr"
        data-testid="input-repo"
        id="repository"
        placeholder="react"
        type="text"
        value="react"
      />
      <div
        class="sc-AxhUy gWaiz"
      >
        <button
          class="sc-AxgMl fHWyBk"
          data-testid="search"
          type="submit"
        >
          Search
        </button>
        <button
          class="sc-AxgMl fHWyBk"
          type="reset"
        >
          Reset
        </button>
      </div>
    </div>
  </form>
</div>
`;
